beamer/state_machine.py

```diff
@@ -63,6 +65,11 @@ class Context:
 
 
 def process_event(event: Event, context: Context) -> HandlerResult:
+    if isinstance(event, SourceChainEvent) and event.chain_id != context.source_chain_id:
+        return True, None
+    if isinstance(event, TargetChainEvent) and event.chain_id != context.target_chain_id:
+        return True, None
```

Comment:
The idea to mark the events via a base type is quite nice. This is really elegant. :+1: